pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -304,8 +304,14 @@ private void readTailMessages(Reader<T> reader) {
                         log.error("Reader {} was closed while reading tail messages.",
                                 reader.getTopic(), ex);
                     } else {
+                        // Retrying on the other exceptions such as NotConnectedException
```

Comment:
Should we push a separate PR and add a unit test for this change?

Reply:
I don't think this is a significant logic change because we are adding a minor delay before retrying(this retrying is a very edge case). If this is a concern, I can make this a separate PR.